project-addons/flask_middleware_connector/events/order_events.py

```diff
@@ -0,0 +1,189 @@
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import models
+from openerp.addons.connector.event import on_record_create, on_record_write, \
+    on_record_unlink
+from openerp.addons.connector.queue.job import job
+from .utils import _get_exporter
+from ..backend import middleware
+from openerp.addons.connector.unit.synchronizer import Exporter
+from ..unit.backend_adapter import GenericAdapter
+
+
+@middleware
+class OrderExporter(Exporter):
+
+    _model_name = ['sale.order']
+
+    def update(self, binding_id, mode):
+        order = self.model.browse(binding_id)
+        vals = {"odoo_id": order.id,
+                "name": order.name,
+                "state": order.state,
+                "partner_id": order.partner_id.id,
+                "amount_total": order.amount_total,
+                "date_order": order.date_order,
+                "amount_untaxed": order.amount_untaxed,
+                "client_order_ref": order.client_order_ref,
+        }
+        if mode == "insert":
+            return self.backend_adapter.insert(vals)
+        else:
+            return self.backend_adapter.update(binding_id, vals)
+
+    def delete(self, binding_id):
+        return self.backend_adapter.remove(binding_id)
+
+@middleware
+class OrderAdapter(GenericAdapter):
+    _model_name = 'sale.order'
+    _middleware_model = 'order'
+
+
+@on_record_create(model_names='sale.order')
+def delay_export_order_create(session, model_name, record_id, vals):
+    order = session.env[model_name].browse(record_id)
+    if order.partner_id.web or order.partner_id.commercial_partner_id.web:
+        if 'state' in vals.keys() and vals['state'] in ('draft', 'reserve'):
+            export_order.delay(session, model_name, record_id, priority=2, eta=80)
+
+
+@on_record_write(model_names='sale.order')
+def delay_export_order_write(session, model_name, record_id, vals):
+    order = session.env[model_name].browse(record_id)
+    #He cogido order_line porque no entra amount_total ni amount_untaxed en el write
+    up_fields = ["name", "state", "partner_id", "date_order", "client_order_ref", "order_line"] 
+    if order.partner_id.web or order.partner_id.commercial_partner_id.web:
+        if 'state' in vals.keys() and vals['state'] not in ('done', 'progress', 'reserve'):
```

Comment:
Yo esto sigo sin verlo así, ahora mismo entre reserve y progress hay otros estados, podéis no enviarlos pero yo no borraría el pedido del middleware sólo lo borarría si viene cancel como valor